stdlib/public/Darwin/Foundation/NSDictionary.swift

```diff
@@ -61,6 +61,78 @@ extension Dictionary : _ObjectiveCBridgeable {
                          to: NSDictionary.self)
   }
 
+  @inline(__always)
+  private static func _typePun<T>(index:Int, of buffer: UnsafePointer<AnyObject>, to: T.Type) -> UnsafeMutablePointer<T> {
+    let typedBase = UnsafeMutableRawPointer(mutating: buffer).assumingMemoryBound(to: T.self)
+    let unboundTarget = UnsafeMutableRawPointer(mutating: typedBase + index)
+    return unboundTarget.bindMemory(to: T.self, capacity: 1)
+  }
+
+  @inline(__always)
+  private static func _verbatimBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) {
+    let verbatimBuffer = UnsafeMutableRawPointer(mutating: objectBuffer).assumingMemoryBound(to:T.self)
+    for i in (0..<count).reversed() {
+      _typePun(index: i, of: objectBuffer, to: T.self).initialize(to: verbatimBuffer[i])
+    }
+  }
+
+  @inline(__always)
+  private static func _nonVerbatimForceBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) {
+    for i in (0..<count).reversed() {
+      let bridged = Swift._forceBridgeFromObjectiveC(objectBuffer[i], T.self)
+      _typePun(index: i, of: objectBuffer, to: T.self).initialize(to: bridged)
+    }
+  }
+  
+  @inline(__always)
+  private static func _nonVerbatimBridge<T>(
+    _ objectBuffer: UnsafePointer<AnyObject>,
+    count: Int,
+    to: T.Type
+  ) -> Bool {
+    for i in (0..<count).reversed() {
+      guard let bridged = objectBuffer[i] as? T else {
+        return false
```

Comment:
On this failure branch, we should release the previously converted elements, or we'll leak them.